scripts/nbconverted/explore_relationship_AandB.py

```diff
@@ -0,0 +1,585 @@
+get_ipython().run_line_magic('autoreload', '2')
+
+import os
+import pandas as pd
+import numpy as np
+import random
+import glob
+import seaborn as sns
+from keras.models import model_from_json, load_model
+from functions import utils
+import matplotlib.pyplot as plt
+from matplotlib_venn import venn2
+
+from numpy.random import seed
+randomState = 123
+seed(randomState)
+
+
+# In[2]:
+
+
+# Run notebook to generate simulated data
+#%run ./main_Pa_sim_enhance_AtoB.ipynb
+
+
+# In[3]:
+
+
+# Load 
+base_dir = os.path.dirname(os.getcwd())
+analysis_name = 'sim_AB_2775_300_v2'
+
+sim_data_file = os.path.join(base_dir, "data", analysis_name, "train_model_input.txt.xz")
+A_file = os.path.join(base_dir, "data", analysis_name, "geneSetA.txt")
+B_file = os.path.join(base_dir, "data", analysis_name, "geneSetB.txt")
+
+offset_vae_file = os.path.join(os.path.dirname(os.getcwd()), "encoded", analysis_name, "offset_latent_space_vae.txt")
+
+weight_file = os.path.join(os.path.dirname(os.getcwd()), "data", analysis_name, "VAE_weight_matrix.txt")
+
+model_encoder_file = glob.glob(os.path.join(base_dir, "models", analysis_name, "*_encoder_model.h5"))[0]
+weights_encoder_file = glob.glob(os.path.join(base_dir, "models", analysis_name, "*_encoder_weights.h5"))[0]
+model_decoder_file = glob.glob(os.path.join(base_dir, "models", analysis_name, "*_decoder_model.h5"))[0]
+weights_decoder_file = glob.glob(os.path.join(base_dir, "models", analysis_name, "*_decoder_weights.h5"))[0]
+
+
+# In[4]:
+
+
+# Read data
+sim_data = pd.read_table(sim_data_file, index_col=0, header=0, compression='xz')
+geneSetA = pd.read_table(A_file, header=0, index_col=0)
+geneSetB = pd.read_table(B_file, header=0, index_col=0)
+
+print(sim_data.shape)
+sim_data.head()
+
+
+# In[5]:
+
+
+# Select samples that have expression of gene A around the threshold 
+# Since threshold is 0.5 then select samples with expression in range(0.4, 0.6)
+
+# Since our simulation set all genes in set A to be the same value for a give sample
+# we can consider a single gene in set A to query by
+rep_gene_A = geneSetA.iloc[0][0]
+
+# Query for samples whose representative gene A expression is in range (0.4, 0.6)
+#test_samples = sim_data.query('0.4 < @rep_gene_A < 0.6') -- why didn't this work?
+test_samples = sim_data[(sim_data[rep_gene_A]>0.4) & (sim_data[rep_gene_A]<0.6)]
+
+test_samples_sorted = test_samples.sort_values(by=[rep_gene_A])
+
+print(test_samples_sorted.shape)
+test_samples_sorted.head()
+
+
+# ## Trend of gene B with respect to A (before transformation)
+# 
+# How is B changing with respect to A in our simulated dataset?
+# 
+# Plot gene expression of A vs mean(gene B expression).  This plot will serve as a reference against the later plot that will show gene expression of A vs mean(transformed gene B expression)
+
+# In[6]:
+
+
+# Get the means of B genes
+
+# Convert dataframe with gene ids to list
+geneSetB_ls = [l[0] for l in geneSetB.values.tolist()]
```

Comment:
Thanks for catching this.  I had been fiddling around trying to use pandas data structures because I didn't think it should be so complex.  When I tried ```geneSetB.values.tolist()``` this returned to me a list of lists so thus the list comprehension.